WordPress/gutenberg
# Move the Switch to Draft button to the Inspector

## What?
This PR moves the "Switch to draft" button from the Top Bar to the visibility panel in the Inspector:

<img width="429" alt="Screenshot 2022-05-10 at 15 57 44" src="https://user-images.githubusercontent.com/846565/167659113-a259a312-c79e-4d05-a0c9-70b2c83d2a4e.png">

## Why?
As mentioned in #40891, the "Switch to Draft" button is perhaps a little prominent given it's a fairly infrequent action. It's also confusing in the template editing context.

## How?
`<PostSwitchToDraftButton />` instance removed from `packages/editor/src/components/post-saved-state/index.js`, and added to `packages/editor/src/components/post-visibility/index.js`.

## Testing Instructions
1. Create a new post.
2. Schedule the post.
3. Open the Visibility panel, and ensure the "Switch to draft" button appears and functions as expected.
4. Publish the post 'immediately'.
5. Open the Visibility panel again, and ensure the "Switch to draft" button appears and functions as expected.
